### PR TITLE
OAuth and Developer Apps QA C-2816 C-2815 C-2794 C-2793 C-2795 C-2797

### PR DESCRIPTION
**Pull request by nicoback2**
### Description

-Fix spacing in developer app list (think the designs were updated)
Screenshot 2023-06-29 at 9 35 15 PM

-Fix name field going red when you click "cancel"
-Don't allow editing fields after submitting
-Add max chars for name field
-Fix logo in OAuth page

-Truncate or word-break long names/descriptions with no spaces to prevent breaking UI
Screenshot 2023-06-29 at 8 33 40 PM

Screenshot 2023-06-29 at 7 13 16 PM

Screenshot 2023-06-29 at 7 13 27 PM

Screenshot 2023-06-29 at 7 02 00 PM

Screenshot 2023-06-29 at 7 08 05 PM

Screenshot 2023-06-29 at 7 00 24 PM

Screenshot 2023-06-29 at 6 59 56 PM

### Dragons

*Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?*

### How Has This Been Tested?
Running web locally against stage

*Please describe the tests that you ran to verify your changes. Provide repro instructions & any configuration.*

### How will this change be monitored?

*For features that are critical or could fail silently please describe the monitoring/alerting being added.*

### Feature Flags ###

*Are all new features properly feature flagged? Describe added feature flags.*

### CODE REVIEW

File: packages/common/src/api/developer-apps.ts [hunk 1]
```diff
@@ -4,12 +4,13 @@ import { ID } from 'models/Identifiers'
...
 
-const descriptionMaxLength = 128
+export const DEVELOPER_APP_DESCRIPTION_MAX_LENGTH = 128
+export const DEVELOPER_APP_NAME_MAX_LENGTH = 50
 
 export const developerAppSchema = z.object({
   userId: z.number(),
-  name: z.string(),
-  description: z.string().max(descriptionMaxLength).optional()
+  name: z.string().max(DEVELOPER_APP_NAME_MAX_LENGTH),
```
Comment (dylanjeffers) [inline packages/common/src/api/developer-apps.ts:12]: Wonderful thanks for doing that

File: packages/web/src/pages/oauth-login-page/OAuthLoginPage.module.css [hunk 2]
```diff
@@ -21,6 +21,9 @@
   font-size: var(--font-l);
   font-weight: var(--font-bold);
   line-height: 140%;
+  width: 100%;
+  overflow: hidden;
```
Comment (dylanjeffers) [inline packages/web/src/pages/oauth-login-page/OAuthLoginPage.module.css:25]: Great call

File: packages/web/src/pages/settings-page/components/desktop/DeveloperApps/DeveloperAppListItem.module.css [hunk 3]
```diff
@@ -8,6 +8,10 @@
   padding: var(--unit-4) var(--unit-6);
 }
 
+.listItem:not(:first-child) {
+  margin-top: var(--unit-2);
```
Comment (dylanjeffers) [inline packages/web/src/pages/settings-page/components/desktop/DeveloperApps/DeveloperAppListItem.module.css:12]: Can we use row gap?

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/nkang--app-name-length

**Review by dylanjeffers — APPROVED** [review]
Awesome

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/nkang--app-name-length

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/nkang--app-name-length

**Pull request merged**